packages/flutter_tools/lib/src/usage.dart

```diff
@@ -49,7 +49,21 @@ const String kCommandBuildBundleTargetPlatform = 'cd24';
 const String kCommandBuildBundleIsModule = 'cd25';
 
 const String kCommandResult = 'cd26';
-// Next ID: cd27
+
+const String kCommandDoctorDeviceValidator = 'cd27';
```

Comment:
The more I think about it the more concerns I have about using analytics this way. We're going to end up with both a lot of missing data if it can't be extensible and "empty" slots if we deprecate/remove/change any of these keys.

Reply:
We have 174 dimensions left. @jayoung-lee  if you get a break-down of the `flutter doctor` result as a JSON object, would you be able to extract the data manually?  The Analytics dashboard won't display this data in a way that is usable for analysis AFAIK.

Reply:
I will be able to retrieve data stored in JSON-formatted strings, using [JSON functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/json_functions) supported in BigQuery. (We are currently exporting the raw data to BigQuery for ad-hoc analysis.)

There are some limitations, though:
1. Those who are using Analytics 360 UI to explore the Analytics data won't be able to use the dimension to visualize the data. 
2. The maximum length of a single custom dimension has to be < 150 bytes, according to this [documentation](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#cd_). 

I agree that we will have empty slots once some measures are deprecated. But from my perspective, I think it will be the same if we deprecate JSON. Furthermore, if we end up changing a part a JSON object, I guess it will be even harder to figure that out from the Analytics 360 UI. Tracking the dimensions independently with unique index for each could be easier to manage.

Reply:
@jayoung-lee If the tool were to report the result as independent events, would that also work? cc @zanderso.

Reply:
Even if sending the result of each check as an independent event is less useful, it would still be my preference to do things that way, as it will greatly simplify this implementation, prevent leaking across abstraction boundaries in this code, and allow new doctor validators to automatically benefit from analytics collection. It looks like additional parameters can also be sent with events to gain back some of the lost ground, if there is any.

Reply:
Yes reporting as events will work too.

Reply:
awesome. I just updated to events.